### PR TITLE
github: publish multi-arc images to hub

### LINKED ISSUE #238: Publish peakcom/s5cmd:v1.1.0 image to docker hub
The peakcom/s5cmd:latest image in docker hub is quite old. Publish a current 1.1.0 image to docker hub.

### PR DESCRIPTION
**Pull request by sonmezonur**
Resolves #238 

Tested it on [my personal hub ](https://hub.docker.com/r/sonmezonur/s5cmd/tags)

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Docker
```
Comment (igungor) [inline .github/workflows/docker.yml:1]: can't we just use goreleaser?

Comment (ilkinulas) [reply .github/workflows/docker.yml:1]: If it is possible to publish releases to github with github actions, my vote is to retire goreleaser and switch to github actions for publishing to docker hub and github.

**Review by ilkinulas — APPROVED** [review]

**Comment by igungor** [comment]
@sonmezonur please rebase for all checks to succeed.

File: .github/workflows/docker.yml (hunk 1 above)
Comment (igungor) [reply .github/workflows/docker.yml:1]: goreleaser is still necessary for cross compilation build+changelog generation.

@sonmezonur mentioned that goreleaser can't publish multiarch docker releases, so we decided to go with docker/build-push-action **and** goreleaser-action.

**Comment by sonmezonur** [comment]
> @sonmezonur please rebase for all checks to succeed.

done

**Review by igungor — APPROVED** [review]

**Pull request merged**